### ISSUE TITLE
Better expm

### ISSUE THREAD

**Issue by howsiyu** [google/jax#17756]
Hi, it seems that `jax.scipy.linalg.expm` 's implementation is based on
"The Scaling and Squaring Method for the Matrix Exponential Revisited"
by Higham, Nicholas J. 2005 which is also featured in "Functions
of Matrices: Theory and Computation". However, scipy's implementation is based on
"A New Scaling and Squaring Algorithm for the Matrix Exponential" by the same author
in 2009 which gives a few improvements; notably, it calculates the diagonal and superdiagonal
of exponential of triangular matrix in exact arithmetic and it uses norms of powers of A to
better determine the number of squarings needed.

Also, I noticed that the current implementation of expm uses pade approximation of degree
3, 5, 7, 9, or 13 depending on the norm of matrix. I believe this is done in the original paper mainly
to save computing power as we can afford to use pade approximation of smaller degree for
matrix of small norm; and that pade approximation of higher degree doesn't hurt the accuracy.
However, in GPU we probably execute all the branches anyway. I wonder whether we should just use
the degree 13 pade approximation, which simplify the code substantially and may help performance
in GPU.

**Comment by jakevdp** [comment]
cc/ @sriharikrishna who first contributed `expm` in #1940
